File: doctor-service/src/main/java/org/example/doctorservice/entity/Department.java
package org.example.doctorservice.entity;

import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;

import java.sql.Timestamp;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "departments")
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Department {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "department_id")
    private Integer departmentId;

    @Column(name = "department_name", length = 100)
    private String departmentName;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "created_at")
    @CreationTimestamp
    private Timestamp createdAt;

    @OneToMany(mappedBy = "department", cascade = CascadeType.ALL, orphanRemoval = true)
    @JsonManagedReference
    private List<ExaminationRoom> examinationRooms = new ArrayList<>();
}

File: frontend/web/src/role-doctor/types/department.ts
import type { ExaminationRoom } from "./examinationRoom";

export interface Department {
  departmentId: number;
  departmentName: string;
  description?: string; 
  location: string;
  headDoctorId?: number; 
  foundedYear?: number; 
  phoneNumber: string;
  email: string;
  createdAt: Date;
  examinationRooms: ExaminationRoom[]; 
}